src/lib/controllers/config-store/adapters/kubernetes-config-store.ts

```diff
@@ -0,0 +1,220 @@
+				}
+			}
+		}
+		return kvPairsToConfigMap(ret);
+	}
+
+	async patchSecret({
+		k8sSecretName,
+		name,
+		value,
+	}: SecretOperationArgs): Promise<void> {
+		const patchManifest = {
+			APIVersion: 'v1',
+			Kind: 'Secret',
+			metadata: {
+				name: k8sSecretName,
+			},
+			type: 'Opaque',
+			data: {
+				[name]: Buffer.from(value).toString('base64'),
+			},
+		};
+		await this.client.api.v1
+			.namespaces(_.get(this.access.kubernetes, 'namespace'))
+			.secrets(k8sSecretName)
+			.patch({ body: patchManifest });
+	}
+
+	async putSecret({
+		k8sSecretName,
+		name,
+		value,
+	}: SecretOperationArgs): Promise<void> {
+		const postManifest = {
+			APIVersion: 'v1',
+			Kind: 'Secret',
+			metadata: {
+				name: k8sSecretName,
+			},
+			type: 'Opaque',
+			data: {
+				[name]: Buffer.from(value).toString('base64'),
+			},
+		};
+		await this.client.api.v1
+			.namespaces(_.get(this.access.kubernetes, 'namespace'))
+			.secrets.post({ body: postManifest });
+	}
+
+	async updateMany(envvars: ConfigMap): Promise<ConfigMap> {
+		if (_.get(this.access.kubernetes, 'bastion')) {
+			return (await runInTunnel(
+				this.tnlConfig,
+				this.updateSecretsConfigMap(envvars),
+				300000,
+			)) as ConfigMap;
+		}
+		return await this.updateSecretsConfigMap(envvars);
+	}
+
+	/**
+	 *
+	 * @param {string} name
+	 * @param {string} value
+	 * @param secrets
+	 * @returns {Promise<boolean>} True if secret exists
+	 */
+	async updateExistingSecret({
+		name,
+		value,
+		secrets,
+	}: UpdateSecretArgs): Promise<boolean> {
+		if (!secrets) {
+			secrets = this.getOpaqueSecrets();
+		}
+
+		for (const secret of secrets) {
+			for (const envvarName of _.keys(secret.data)) {
+				if (
+					envvarName === name &&
+					value.toString() !==
+						Buffer.from(secret.data[envvarName], 'base64').toString()
+				) {
+					await this.patchSecret({
+						k8sSecretName: secret.metadata.name,
+						name: envvarName,
+						value,
+					});
+					return true;
+				}
+				if (envvarName === name) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	async updateSecret({
+		name,
+		value,
+		secrets,
+	}: UpdateSecretArgs): Promise<void> {
+		if (!secrets) {
+			secrets = this.getOpaqueSecrets();
+		}
+		const exists = await this.updateExistingSecret({ name, value, secrets });
+		if (!exists) {
+			const k8sSecretName = name.toLowerCase().replace(/_/g, '-');
+			await this.putSecret({ k8sSecretName, name, value });
+		}
+	}
+
+	async getOpaqueSecrets(): Promise<any> {
+		const secrets = await this.client.api.v1
+			.namespaces(_.get(this.access.kubernetes, 'namespace'))
+			.secrets.get();
+		return _.filter(secrets.body.items, { type: 'Opaque' });
+	}
+
+	async updateSecretsConfigMap(envvars: ConfigMap): Promise<ConfigMap> {
+		const secrets = await this.getOpaqueSecrets();
+		const envvarPairs = configMapToPairs(envvars);
+		for (const name of _.keys(envvarPairs)) {
+			await this.updateSecret({
+				name,
+				value: envvarPairs[name].toString(),
```

Comment:
There's an awful lot of `.toString()` over the entire config. Would it be possible to convert this. Why not `toString()` all pairs on load?